base/pcre.jl

```diff
@@ -1,19 +1,25 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
-## low-level pcre interface ##
+## low-level pcre2 interface ##
 
 module PCRE
 
 include("pcre_h.jl")
 
-const VERSION = bytestring(ccall((:pcre_version, :libpcre), Ptr{UInt8}, ()))
 
 global JIT_STACK = C_NULL
+global MATCH_CONTEXT = C_NULL
+
 function __init__()
     JIT_STACK_START_SIZE = 32768
     JIT_STACK_MAX_SIZE = 1048576
-    global JIT_STACK = ccall((:pcre_jit_stack_alloc, :libpcre), Ptr{Void},
-                             (Cint, Cint), JIT_STACK_START_SIZE, JIT_STACK_MAX_SIZE)
+    global JIT_STACK = ccall((:pcre2_jit_stack_create_8, "libpcre2-8"), Ptr{Void},
```

Comment:
Is this `libpcre2-8` likely to change in the future? If so, would be best to make a single const variable for it.